src/components/side_panel/data_validation/dv_editor/dv_editor.xml

```diff
@@ -23,6 +23,7 @@
             t-key="state.rule.criterion.type"
             criterion="state.rule.criterion"
             onCriterionChanged.bind="onCriterionChanged"
+            autofocus="state.isTypeUpdated"
```

Comment:
shouldn't the t-key already force the destroy/reload of the component? 
I think autofocus="true" could do the trick